doc/conf.py

```diff
@@ -242,6 +242,14 @@
 linkcheck_retries = 3
 linkcheck_timeout = 500
 
+
+# Pages are not detected correct by ``make linkcheck``
+linkcheck_ignore = [
+    'https://data.kitware.com/#collection/55f17f758d777f6ddc7895b7/folder/5afd932e8d777f15ebe1b183',
+    'https://www.sciencedirect.com/science/article/abs/pii/S0309170812002564',
+    'https://www.researchgate.net/publication/2926068_LightKit_A_lighting_system_for_effective_visualization',
+]
```

Comment:
Do you know why these are not detected as correct? All three links seem to work for me in a browser. Do issues arise in CI? Or with the user agent being used by sphinx when making requests?

Reply:
Here is what is reported when running linkcheck against those files:

```
broken    2926068_LightKit_A_lighting_system_for_effective_visualization - 

broken    https://data.kitware.com/#collection/55f17f758d777f6ddc7895b7/folder/5afd932e8d777f15ebe1b183 - Anchor 'collection/55f17f758d777f6ddc7895b7/folder/5afd932e8d777f15ebe1b183' not found

broken    https://www.researchgate.net/publication/ - 403 Client Error: Forbidden for url: https://www.researchgate.net/publication/
```

I've seen this before and this was the solution, what are your thoughts on this?

Reply:
Thanks. That's a mixture of weird and understandable, then.

```
broken    https://data.kitware.com/#collection/55f17f758d777f6ddc7895b7/folder/5afd932e8d777f15ebe1b183 - Anchor 'collection/55f17f758d777f6ddc7895b7/folder/5afd932e8d777f15ebe1b183' not found
```
This is fair enough: it seems the website is abusing anchor syntax in the URL to encode paths in there. I don't really know web things, but `https://some.domain/some/path#thing` should refer to something on page `https://some.domain/some/path` so it's probably fine for linkcheck to be confused. Unless we can disable linkcheck complaining about URL fragments, it's fine to ignore this here.

Now the others:
```
broken    2926068_LightKit_A_lighting_system_for_effective_visualization - 
broken    https://www.researchgate.net/publication/ - 403 Client Error: Forbidden for url: https://www.researchgate.net/publication/
```

These are weird: it looks as if the one URL was broken up into two? Sounds weird. Oddly enough the only URL for this I can find is here:
https://github.com/pyvista/pyvista/blob/b5d983a044747355fa8ea2e3ab5b6d4edbc60c0c/pyvista/plotting/plotting.py#L935-L936

Perhaps linkcheck is having issues with the redirect from this to what you added to ignore?

Reply:
Oh, and your last comment didn't include https://www.sciencedirect.com/science/article/abs/pii/S0309170812002564. Is the issue intermittent?

Reply:
I said
> Oddly enough the only URL for this I can find is here:

But because I was looking at `main` rather than this PR, sorry. Then I don't understand why that gets broken up... In any case 403 error might mean that researchgate doesn't like linkcheck trying to download something (anti-scraping measure, I assume).

Reply:
Ops, I was still fixing the conflicts. You are correct cause is the way the URL is written which is not correct and linkcheck fails to check. This seems to be a known issue at least with the GitHub pages and how they make the anchor. 

https://github.com/sphinx-doc/sphinx/issues/9435

Before ignoring the links, I tried to see if I could find the right anchor of the link. So those are the ones that I did not figured out; therefore, I propose to add in the ignore. This comes with the risk that those links may go dead at some point and it goes unnoticed. However, I think is a workaround for now. Wdyt?

I have tested with researchgate again with the following:


```
broken    https://www.researchgate.net/publication/2926068 - 403 Client Error: Forbidden for url: https://www.researchgate.net/publication/2926068
```

Also, reports as broken.

Reply:
If I make a request to https://www.researchgate.net/publication/2926068 with `curl` I get this:
![Screenshot from 2022-08-20 18-04-43](https://user-images.githubusercontent.com/17914410/185756012-c6e33f6f-4328-4e3b-92ad-a97fe9f74b13.png)

Same with `requests`:
```py
>>> import requests
>>> requests.get('https://www.researchgate.net/publication/2926068').status_code
403
```

So we're indeed stuck with ignoring this.